h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
+
+    var getActiveElementIndex = () => {
+      return Array.prototype.slice.call(this._dropdownItems).
+        indexOf(element.querySelector('.js-search-bar-dropdown-menu-item--active'));
```

Comment:
Ah, and @robertknight points out that there's also [Array.from()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)